File: TsogosunProfileAdminWeb/src/app/service/profile-rewards-admin/communication-preference.service.ts
import { Injectable } from '@angular/core';
import { CommunicationPreferenceDto } from 'src/app/model/Dtos/profile-rewards-admin/communicationPreferenceDto';
import { BackendService } from 'src/app/util/backend.service';

@Injectable({
    providedIn: 'root'
})
export class CommunicationPreferenceService {

    constructor(private backend: BackendService) {

    }

    async getAllCommunicationPreferencesByPatron(patronNo: string, siteId: string) {
        return await this.backend.get<CommunicationPreferenceDto[]>(`CommunicationPreference/All/${patronNo}/${siteId}`);
    }    
}

File: TsogosunProfileAdmin/MSPatronRewardsAdmin/Service/CommunicationPreferenceService.cs


using MSPatronRewardsAdmin.Model.Dtos;
using MSPatronRewardsAdmin.Repository.Interface;
using MSPatronRewardsAdmin.Service.Interface;
using System.Collections.Generic;

namespace MSPatronRewardsAdmin.Service
{
    public class CommunicationPreferenceService : ICommunicationPreferenceService
    {

        private readonly ICommunicationPreferenceRepository _communicationPreferenceRepository;

        public CommunicationPreferenceService(ICommunicationPreferenceRepository communicationPreferenceRepository)
        {
            _communicationPreferenceRepository = communicationPreferenceRepository;
        }

        public List<CommunicationPreferenceDto> GetAllCommunicationPreferencesBySiteId(string patronNo, int siteId)
        {
            return _communicationPreferenceRepository.GetAllCommunicationPreferencesBySiteId(patronNo, siteId);
        }
    }
}
